fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java

```diff
@@ -1259,14 +1259,27 @@ public void createTableLike(CreateTableLikeStmt stmt) throws DdlException {
             } finally {
                 table.readUnlock();
             }
-            CreateTableStmt parsedCreateTableStmt = (CreateTableStmt) SqlParserUtils.parseAndAnalyzeStmt(
-                    createTableStmt.get(0), ConnectContext.get());
-            parsedCreateTableStmt.setTableName(stmt.getTableName());
-            parsedCreateTableStmt.setIfNotExists(stmt.isIfNotExists());
-            createTable(parsedCreateTableStmt);
+            ConnectContext ctx = ConnectContext.get();
+            try {
+                // analyze CreateTableStmt will check create_priv of existedTable, create table like only need
+                // create_priv of newTable, and select_priv of existedTable, and priv check has done in
+                // CreateTableStmt/CreateTableCommand, so we skip it
+                if (ctx != null) {
+                    ctx.setSkipAuth(true);
+                }
+                CreateTableStmt parsedCreateTableStmt = (CreateTableStmt) SqlParserUtils.parseAndAnalyzeStmt(
+                        createTableStmt.get(0), ConnectContext.get());
```

Comment:
could parseAndAnalyzeStmt  work well when `ConnectContext.get()` is null ?
```suggestion
                CreateTableStmt parsedCreateTableStmt = (CreateTableStmt) SqlParserUtils.parseAndAnalyzeStmt(
                        createTableStmt.get(0), ctx);
```